cmd/thanos/sidecar.go

```diff
@@ -490,16 +496,23 @@ func (s *promMetadata) UpdateLabels(ctx context.Context) error {
 	return nil
 }
 
-func (s *promMetadata) UpdateTimestamps(mint, maxt int64) {
+func (s *promMetadata) UpdateTimestamps(ctx context.Context) error {
 	s.mtx.Lock()
 	defer s.mtx.Unlock()
 
+	mint, err := s.client.LowestTimestamp(ctx, s.promURL)
+	if err != nil {
+		return err
+	}
+
 	if mint < s.limitMinTime.PrometheusTimestamp() {
```

Comment:
While going through this function we can probably update to use the `min()` builtin

Reply:
Oh yeah, good point; ill update in a bit!